### ISSUE TITLE
Files are not resampled to 48 kHz before calculating NISQA

### ISSUE THREAD

**Issue by philgzl** [urgent-challenge/urgent2024_challenge#13]
Correct me if I am wrong but it seems the files are not resampled to 48 kHz when calculating NISQA. This is because when initializing the `NL.SpeechQualityDataset`, the sampling rate is set to `model.args["ms_sr"]`, which is `None` in [nisqa.tar](https://github.com/gabrielmittag/NISQA/blob/master/weights/nisqa.tar).

https://github.com/urgent-challenge/urgent2024_challenge/blob/478465e37ecd27d1fbca4810aaf4bab7fd6353fe/evaluation_metrics/nisqa_utils.py#L124-L128

As a consequence, the files are not resampled when calling `librosa.load` [here](https://github.com/gabrielmittag/NISQA/blob/ac831378483aa75876c1147acb04b104f6f1d10c/nisqa/NISQA_lib.py#L2299), despite the NISQA metric being described as running at 48 kHz and the comment below.

https://github.com/urgent-challenge/urgent2024_challenge/blob/478465e37ecd27d1fbca4810aaf4bab7fd6353fe/evaluation_metrics/nisqa_utils.py#L106-L110

**Comment by Emrys365** [comment]
@philgzl Thanks for your report! Indeed the default value of `ms_sr` (48000) in [NISQA_lib.SpeechQualityDataset](https://github.com/gabrielmittag/NISQA/blob/master/nisqa/NISQA_lib.py#L2074) will be overwritten by `model.args["ms_sr"]`, which turns out to be `None` for all the pretrained model weights in https://github.com/gabrielmittag/NISQA/tree/master/weights.

However, this behavior inherently exists in the original repository [gabrielmittag/NISQA](https://github.com/gabrielmittag/NISQA). Since we are strictly following the original repository's implementation in [run_predict.py](https://github.com/gabrielmittag/NISQA/blob/master/run_predict.py) → [nisqa/NISQA_model.py](https://github.com/gabrielmittag/NISQA/blob/master/nisqa/NISQA_model.py#L755),
```python
        self.ds_val = NL.SpeechQualityDataset(
            df_val,
            df_con=None,
            data_dir = self.args['data_dir'],
            filename_column = 'deg',
            mos_column = 'predict_only',              
            seg_length = self.args['ms_seg_length'],
            max_length = self.args['ms_max_segments'],
            to_memory = None,
            to_memory_workers = None,
            seg_hop_length = self.args['ms_seg_hop_length'],
            transform = None,
            ms_n_fft = self.args['ms_n_fft'],
            ms_hop_length = self.args['ms_hop_length'],
            ms_win_length = self.args['ms_win_length'],
            ms_n_mels = self.args['ms_n_mels'],
            ms_sr = self.args['ms_sr'],
            ms_fmax = self.args['ms_fmax'],
            ms_channel = self.args['ms_channel'],
            double_ended = self.args['double_ended'],
            dim = self.args['dim'],
            filename_column_ref = None,
            )
```

it should yield consistent results if you run the evaluation script provided in the original repository.

Anyway, this is an important finding, and we encourage you to post an issue in the [gabrielmittag/NISQA](https://github.com/gabrielmittag/NISQA) repository.

**Comment by philgzl** [comment]
Thanks!

Actually upon further investigation it seems this is the normal behavior. As Gabriel explained [here](https://github.com/gabrielmittag/NISQA/issues/6#issuecomment-838157571), the window and hop lengths are adjusted depending on the sampling rate, such that they are always 20 ms and 10 ms. The FFT length is then always 4096, and the maximum mel frequency is always 20 kHz. So signals with low sampling rates end up with empty mel bins at high frequencies, which should result in a lower score compared to fullband signals.

So the only real issue, if anything, is to believe the metric runs at 48 kHz. When reading the paper, it does not seem like the metric was advertised as such (although the metric does "expect" a fullband signal, in the sense that narrowband signals are penalized).

**Comment by Emrys365** [comment]
I see. Thanks for the investigation. Indeed, the resampled window and hop lengths can be an alternative way to handle different sample rates, as also indicated by the baseline framework in the URGENT Challenge.

We might not say that it always runs at 48 kHz, but that it somewhat generalizes to other sample rates because of the above operation. Meanwhile, it is mentioned in the [NISQA paper](https://arxiv.org/abs/2104.09494) that

> The presented model is focused on distortions that occur in modern speech communication networks. However, the model weights can also be used to fine-tune the model for related tasks, such as the prediction of enhanced or synthesised speech.


Therefore,  we still need to find better metrics that fit well the speech enhancement task and generalize well. This is also one of the goals of the URGENT Challenge.
